qiskit/transpiler/passes/mapping/stochastic_swap.py

```diff
@@ -98,18 +86,14 @@ def run(self, dag):
             compatible with the DAG
         """
 
-        if self.initial_layout is None:
-            if self.property_set["layout"]:
-                self.initial_layout = self.property_set["layout"]
-            else:
-                self.initial_layout = Layout.generate_trivial_layout(*dag.qregs.values())
+        if len(dag.qregs) != 1 or dag.qregs.get('q', None) is None:
+            raise TranspilerError('Basic swap runs on physical circuits only')
 
-        if len(dag.qubits()) != len(self.initial_layout):
+        if len(dag.qubits()) > len(self.coupling_map.physical_qubits):
             raise TranspilerError('The layout does not match the amount of qubits in the DAG')
 
-        if len(self.coupling_map.physical_qubits) != len(self.initial_layout):
-            raise TranspilerError(
-                "Mappers require to have the layout to be the same size as the coupling map")
+        canonical_register = dag.qregs['q']
+        self.initial_layout = Layout.generate_trivial_layout(canonical_register)
```

Comment:
Same as for the `BasicSwap`, I get that the mappers need to keep record of the downstream layout as it is modified, but we should pick a different name so as to avoid confusion with a user specified or pass selected `initial_layout` from the property set.

Reply:
I did the same as basic swap here called it trivial layout.